typing/printtyp.ml

```diff
@@ -2458,10 +2458,15 @@ let explanation (type variety) intro prev env
       | _ -> ignore
     in
     explain_escape pre kind
-  | Errortrace.Incompatible_fields { name; _ } ->
-      Some(dprintf "@,Types for method %a are incompatible"
-             Style.inline_code name
-          )
+  | Errortrace.Incompatible_fields { name; diff} ->
+    reserve_names diff.got;
+    reserve_names diff.expected;
+    Some(dprintf "@,@[Types for method %a are incompatible, the method has \
+                    type@ %a,@ but the expected method type was@ %a@]"
```

Comment:
The message here is now very similiar to the other `Incompatible_fields` case above (currently on lines 2451-2457.) Could you share code between the two with a new `explain_incompatible_fields` helper function? (This means producing the same message in both cases, which is also nice for consistency. Personally I think that the message above is slightly better as it says the same thing with less words.)

Reply:
Done in https://github.com/ocaml/ocaml/pull/12777/commits/b01dfaeb432d5b63bf19109ae66fa363e7903cdc.